backbone/src/main/java/org/researchstack/backbone/step/active/recorder/LocationRecorder.java

```diff
@@ -0,0 +1,153 @@
+     * @param minDistance per Android doc, minimum distance between location updates, in meters, no minimum if zero
+     * @param identifier the recorder's identifier
+     * @param step the step that contains this recorder
+     * @param outputDirectory the output directory of the file that will be written with location data
+     */
+    LocationRecorder(long minTime, float minDistance, String identifier, Step step, File outputDirectory) {
+        super(identifier, step, outputDirectory);
+        this.minTime = minTime;
+        this.minDistance = minDistance;
+    }
+
+    @Override
+    public void start(Context context) {
+
+        if (locationManager == null) {
+            locationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+        }
+
+        // In Android, you can register for both network and gps location updates
+        // Let's just register for both and log all locations to the data file
+        // with their corresponding accuracy and other data associated
+
+        try {
+            locationManager.requestLocationUpdates(
+                    LocationManager.NETWORK_PROVIDER, minTime, minDistance, this);
+        } catch (java.lang.SecurityException ex) {
+            Log.i(TAG, "fail to request location update, ignore", ex);
+        } catch (IllegalArgumentException ex) {
+            Log.d(TAG, "network provider does not exist, " + ex.getMessage());
+        }
+
+        try {
+            locationManager.requestLocationUpdates(
+                    LocationManager.GPS_PROVIDER, minTime, minDistance, this);
+        } catch (java.lang.SecurityException ex) {
+            Log.i(TAG, "fail to request location update, ignore", ex);
+        } catch (IllegalArgumentException ex) {
+            Log.d(TAG, "gps provider does not exist " + ex.getMessage());
+        }
+
+        coordinateJsonObject = new JsonObject();
+        jsonObject = new JsonObject();
+        startJsonDataLogging();
+    }
+
+    /**
+     * Ignore the missing permission, since user's of this SDK don't always need it,
+     * unless they are specifically doing a walking task test
+     */
+    @Override
+    public void stop() {
+        if (locationManager != null) {
+            try {
+                locationManager.removeUpdates(this);
+            } catch (Exception ex) {
+                Log.i(TAG, "fail to remove location listners, ignore", ex);
+            }
+        }
+        stopJsonDataLogging();
+    }
+
+    // LocationListener methods
+
+    @Override
+    public void onLocationChanged(Location location) {
+        if (location != null) {
+            jsonObject.addProperty(TIMESTAMP_KEY, location.getTime());
+
+            coordinateJsonObject.addProperty(LONGITUDE_KEY, location.getLongitude());
+            coordinateJsonObject.addProperty(LATITUDE_KEY, location.getLatitude());
+            jsonObject.add(COORDINATE_KEY, coordinateJsonObject);
+
+            if (location.hasAccuracy()) {
```

Comment:
Is it worth rejecting updates entirely if the accuracy is > a threshold?  Or I suppose that could be done in the data analysis.